src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -55,10 +69,47 @@ private[impl] class LaunchQueueActor(
   /** ActorRefs of the actors have been currently suspended because we wait for their termination. */
   var suspendedLaunchersMessages = Map.empty[ActorRef, Vector[DeferredMessage]].withDefaultValue(Vector.empty)
 
+  private var updatesByRunSpecId =
+    Map.empty[PathId, Queue[QueuedAdd]].withDefaultValue(Queue.empty)
```

Comment:
`private[this]` usually preferred for mutable data structures.

Also is there any way a failure could cause us to accumulate entries here?

Reply:
I don't understand. Could you be a little more specific. What do you mean by accumulate entries?